premium/usage.go

```diff
@@ -361,3 +357,33 @@ func (u *BatchUpdater) calculateRetryDuration(statusCode int, headers http.Heade
 func retryableStatusCode(statusCode int) bool {
 	return statusCode == http.StatusTooManyRequests || statusCode == http.StatusServiceUnavailable
 }
+
+func (u *BatchUpdater) getTeamNameByTokenType(tokenType auth.TokenType) (string, error) {
+	switch tokenType {
+	case auth.BearerToken:
+		teamName, err := config.GetValue("team")
+		if errors.Is(err, os.ErrNotExist) {
+			return "", fmt.Errorf("config file for reading team name not found (%w). Hint: use `cloudquery login` and/or `cloudquery switch <team>`", err)
+		} else if err != nil {
+			return "", fmt.Errorf("failed to get team name from config: %w", err)
+		}
+		if teamName == "" {
+			return "", fmt.Errorf("team name not set. Hint: use `cloudquery switch <team>`")
+		}
+		return teamName, nil
+	case auth.APIKey:
+		resp, err := u.apiClient.ListTeamsWithResponse(context.Background(), &cqapi.ListTeamsParams{})
```

Comment:
The use of `context.Background()` isn't very nice here, but then it's in a constructor so the constructor would have to get a ctx otherwise.

Reply:
Agreed, but considered it better than changing the constructor in this case.